File: agri-analysis-client/src/api/category.js
import request from '@/utils/request';

export async function getProvinceList() {
  const { data } = await request({
    method: 'get',
    url: '/api/province'
  });
  return data;
}

export async function getCityList(provinceId) {
  const { data } = await request({
    method: 'get',
    url: '/api/city?province=' + provinceId
  });
  return data;
}

export async function getMarketList(cityId) {
  const { data } = await request({
    method: 'get',
    url: '/api/market?city=' + cityId
  });
  return data;
}

export async function getMarket(marketId) {
  const { data } = await request({
    method: 'get',
    url: '/api/market/' + marketId
  });
  return data;
}

export async function getTypeList() {
  const { data } = await request({
    method: 'get',
    url: '/api/type'
  });
  return data;
}

export async function getVarietyList(typeId) {
  const { data } = await request({
    method: 'get',
    url: '/api/variety?type=' + typeId
  });
  return data;
}

export async function getVariety(varietyId) {
  const { data } = await request({
    method: 'get',
    url: '/api/variety/' + varietyId
  });
  return data;
}

File: agri-analysis-server/app/routes/category.py
from flask import Blueprint, jsonify, request

from app.models.city import City
from app.models.market import Market
from app.models.province import Province
from app.models.type import Type
from app.models.variety import Variety
from app.utils.serialize import serialize

bp = Blueprint('category_bp', __name__, url_prefix='/')


@bp.route('/province', methods=['GET'])
def get_province_list():
    return jsonify(serialize(Province.query.all()))


@bp.route('/city', methods=['GET'])
def get_city_list():
    province_id = request.args['province']
    return jsonify(serialize(City.query.filter_by(province_id=province_id).all()))


@bp.route('/market', methods=['GET'])
def get_market_list():
    city_id = request.args['city']
    return jsonify(serialize(Market.query.filter_by(city_id=city_id).all()))


@bp.route('/market/<market_id>', methods=['GET'])
def get_market(market_id):
    return jsonify(serialize(Market.query.filter_by(id=market_id).one()))


@bp.route('/type', methods=['GET'])
def get_type_list():
    return jsonify(serialize(Type.query.all()))


@bp.route('/variety', methods=['GET'])
def get_variety_list():
    type_id = request.args['type']
    return jsonify(serialize(Variety.query.filter_by(type_id=type_id).all()))


@bp.route('/variety/<variety_id>', methods=['GET'])
def get_variety(variety_id):
    return jsonify(serialize(Variety.query.filter_by(id=variety_id).one()))
